rust/core-lib/src/plugins/manager.rs

```diff
@@ -145,6 +146,41 @@ impl PluginManager {
         }
     }
 
+    fn request_plugins_sync(&self, view_id: ViewIdentifier,
+                            only_globals: bool, method: &str, params: &Value)
+        -> Vec<Result<Value, xi_rpc::Error>>
+    {
+        let num_global_plugins = self.global_plugins.len();
+        let num_non_global_plugins = {
+            if only_globals {
+                0
+            } else if let Ok(locals) = self.running_for_view(view_id) {
+                locals.len()
+            } else {
+                0
+            }
+        };
+        let num_plugins = num_global_plugins + num_non_global_plugins;
+        let mut gathered_results = Vec::with_capacity(num_plugins);
+
+        // TODO: query plugins in parallel & gather results.
```

Comment:
Not sure if you saw, but there _is_ a non-blocking request method that takes a callback. Or at least, there used to be. This would require something like using an atomic int type to track how many requests you send, and then decrement that when each returns, firing some finalizer when it hits zero... probably more bookkeeping than is worth doing right now, but at least an option.

Reply:
Yeah i was concerned about whether it would deadlock (wasn't sure about which thread it would come back on) which is why I went the synchronous route.  I was going to use a condition variable with a vector although the conditional variable has a pretty bad API at the moment for conditional wait (I have a CL out for stdlib to add them back but currently it's unstable & not submitted yet).